build.gradle

```diff
@@ -23,12 +23,20 @@ buildScan {
 subprojects {
     apply plugin: 'java'
     apply plugin: 'maven'
+    apply plugin: 'checkstyle'
 
     repositories {
         mavenLocal()
         mavenCentral()
     }
 
+    checkstyle {
+        // Stay on checkstyle version 6 until Java version 7 support can be dropped
+        toolVersion = '6.19'
+        showViolations = true
+        configFile = file("${project.rootDir}/checkstyle.xml")
```

Comment:
May wanna add suppression file support until all existing code can be converted.   See: https://gist.github.com/soucekv/a962f788ceb70e418e2f

In my current project, I basically used a script to add *every* file to the suppression script and then just have a standing rule that team members to try and remove files from the suppression list whenever they need to significantly modify a file.  That at least keeps the problem from getting worse as new files are added, but also encourages devs to help clean up the old stuff when they have the chance.